daemon/api.go

```diff
@@ -362,8 +363,11 @@ func searchStore(c *Command, r *http.Request, user *auth.UserState) Response {
 		return InternalError("cannot find route for snaps")
 	}
 
-	query := r.URL.Query()
+	if r.URL.Query().Get("select") == "refresh" {
+		return storeUpdates(c, r)
+	}
 
+	query := r.URL.Query()
```

Comment:
This may be moved above the conditional block so we can already use it there.

Also, is there an easy way to respect "q", given that we're asking the store for these updates?

If not, we need to reject queries that provide both, rather than silently doing something else.